File: 467-unique-substrings-in-wraparound-string/467-unique-substrings-in-wraparound-string.py
class Solution:
    def findSubstringInWraproundString(self, p: str) -> int:
        d = defaultdict(int)
        d[p[0]] = 1
        l = 1
        for i in range(1,len(p)):
            if(p[i] == 'a' and p[i-1] == 'z' or p[i] == chr(ord(p[i-1])+1)):
                l += 1
            else:
                l = 1
            d[p[i]] = max(l,d[p[i]])
                
        return sum(d.values())

File: 467-unique-substrings-in-wraparound-string/467-unique-substrings-in-wraparound-string.cpp
class Solution {
public:
    int findSubstringInWraproundString(string p) {
       int n = p.length();
        
        map<char,int> mp;
        mp[p[0]] = 1;
        int len = 1;
        for(int i=1;i<n;i++){
            if(p[i] == p[i-1]+1 || p[i]=='a' && p[i-1]=='z'){
                len++;
            }
            else len = 1;
            
            mp[p[i]] = max(mp[p[i]],len);
        }
        int sum = 0;
        for(auto &it: mp) sum += it.second;
        return sum;
    }
};